tests/test_big_modeling.py

```diff
@@ -370,11 +371,17 @@ def test_dispatch_model_tied_weights(self):
         dispatch_model(model, device_map)
         assert model.linear2.weight is model.linear1.weight
 
-    @require_multi_gpu
+    @require_multi_gpu_or_xpu
     def test_dispatch_model_tied_weights_memory(self):
         # Test that we do not duplicate tied weights at any point during dispatch_model call.
 
-        torch.cuda.empty_cache()  # Needed in case we run several tests in a row.
+        torch_accelerator_module = None
+        if torch_device_type == "cuda":
+            torch_accelerator_module = torch.cuda
+        elif torch_device_type == "xpu":
+            torch_accelerator_module = torch.xpu
```

Comment:
This can just be:

```suggestion
        torch_accelerator_module = getattr(torch, torch_device_type)
```

And this simplifies much of the downstream code